### ISSUE TITLE
Javascript debugger does not connect to headless instance

### ISSUE THREAD

**Issue by 5hee75** [microsoft/vscode-edge-devtools#896]
**Environment (please complete the following information):**
 - OS: MacOS 12.1 (M1 Mac Mini)
 - Extension version: 1.4.5

**Describe the bug:**

The Javascript debugger does not connect to headless Edge instance.

**Repro steps:**

* Launch web server with npm in debug shell
* Launch edge devtools in headless mode with screencast
* Debugger does not attach and breakpoints remain disabled

**Expected behavior:**

Debugger should attach to the instance of Edge running headless with screencast, just like it does with a separate Edge window.

**Additional context:**

I am attempting to launch my app using a compound configuration in `launch.json` (see below for specifics).  If I launch Edge using `type: pwa-msedge` then my web server debugger attaches just fine and breakpoints work as expected.  I can also attach the Edge DevTools in VSCode to that running instance.

However, I would ideally like a completely unified dev experience, with code, dev tools, and browser embedded within a single VSCode window.  I can launch an independent, headless instance of Edge with screencast in my VSCode window, but launching the DevTools directly in this manner does not connect to my app's debugger/breakpoints.

Snippets from `launch.json`

```json
"configurations": [
    {
      "name": "Open App",
      "request": "launch",
      "type": "pwa-msedge",
      "url": "http://localhost:3000",
      "webRoot": "${workspaceFolder}",
    },
    {
      "name": "Open App (Headless)",
      "request": "launch",
      "type": "vscode-edge-devtools.debug",
      "url": "http://localhost:3000",
      "webRoot": "${workspaceFolder}",
    },
    {
      "name": "Start App",
      "command": "npm run start",
      "request": "launch",
      "type": "node-terminal"
    }
],
"compounds": [
    {
      "name": "Debug App (Works)",
      "configurations": ["Start App", "Open App"]
    },
    {
      "name": "Debug App (Does not work)",
      "configurations": ["Start App", "Open App (Headless)"]
    }
]
```

**Comment by codepo8** [comment]
You can add the headless option to your launch.json and it works (at least here right now)

```javascript
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "pwa-msedge",
            "request": "launch",
            "name": "webdebug",
            "url": "http://localhost:8080",
            "webRoot": "${workspaceFolder}",
            "runtimeExecutable": "stable",
            "runtimeArgs": ["--headless"]
        }
    ]
}
```

**Comment by chris-miner** [comment]
> If I launch Edge using `type: pwa-msedge` then my web server debugger attaches just fine and breakpoints work as expected. I can also attach the Edge DevTools in VSCode to that running instance.

I had this same issue recently, and was directed by @codepo8 to have a look at https://docs.microsoft.com/en-us/microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension#browser-debugging-with-microsoft-edge-devtools-integration-in-visual-studio-code

Wherein I discovered that the behavior you describe is apparently how it is supposed to work.  If you want to debug javascript, you need to do the pwa-msedge launch.  If you want to fix up display issues, use the vscode-edge-devtools.debug.

**Comment by codepo8** [comment]
> > If I launch Edge using `type: pwa-msedge` then my web server debugger attaches just fine and breakpoints work as expected. I can also attach the Edge DevTools in VSCode to that running instance.
> 
> I had this same issue recently, and was directed by @codepo8 to have a look at https://docs.microsoft.com/en-us/microsoft-edge/visual-studio-code/microsoft-edge-devtools-extension#browser-debugging-with-microsoft-edge-devtools-integration-in-visual-studio-code
> 
> Wherein I discovered that the behavior you describe is apparently how it is supposed to work. If you want to debug javascript, you need to do the pwa-msedge launch. If you want to fix up display issues, use the vscode-edge-devtools.debug.

Not quite - and that's something I probably need to fix in the docs. You can always use pwa-msedge, which triggers the JavaScript debugging. In the debugging toolbar you can then launch the tools for visual fixes by activating the devtools icon: 

![image](https://user-images.githubusercontent.com/13508/155207979-80f2e151-fdab-4acd-bb83-01d4b6689a70.png)

There is a step-by-step article here: https://dev.to/codepo8/a-launchjson-to-turn-vs-code-into-a-full-web-development-ide-36ge

**Comment by chris-miner** [comment]
> In the debugging toolbar you can then launch the tools for visual fixes by activating the devtools icon:

I think that is clear from the docs.  That's how I've been launching the dev tools.  Although, the compound launch options outline above are a real workflow boost.

**Comment by codepo8** [comment]
I filed this as a bug to follow up. I agree that as a compound this would be a boost to productivity.

**Comment by 5hee75** [comment]
> You can add the headless option to your launch.json and it works (at least here right now). Notice the runtimeArgs.

This is nice workaround for the time being.  If I include both the headless `pwa-msedge` as well as the `vscode-edge-devtools.debug` then it effectively behaves as desired.  Thanks!

**Comment by mliao95** [comment]
We have updated our `launch.json` configuration to include updated workflows like the ones outlined in this issue.  If you want to try it out, click on "Configure launch.json" when the targets list is empty on the extension sidebar.  This should auto-generate useful debug configurations to launch your project and connect to it with the DevTools

![image](https://user-images.githubusercontent.com/14304598/163052407-522f1436-dc2e-4bde-a677-276b102ee2bd.png)

Closing out this issue.
